### PR TITLE
Support Swift 2.3 and Xcode 8

### LINKED ISSUE #1440: Compatibility with Xcode 8
It seems like we've lost compatibility with Xcode 8. We should fix that so Carthage will build correctly on macOS Sierra.

### PR DESCRIPTION
**Pull request by ikesyo**
Resolves #1440. This still preserves Swift 2.2/Xcode 7.3 compatibility.

- [x] Fix test failures:
    - 3f22133: https://github.com/ikesyo/carthage-fixtures-ReactiveCocoaLayout
    - ecfcd52, ad6be03: https://github.com/ikesyo/CarthageCopyFrameworksFixture/releases/tag/1.1.0
    - 611105c
    - 89c421b: We are hitting a weird issue related to running a UI test with Xcode 8 beta on Travis. So let's work around that: https://github.com/travis-ci/travis-ci/issues/6519
- [x] Update `PrettyColors` to a tagged version: https://github.com/jdhealy/PrettyColors/pull/12

### CODE REVIEW

File: Source/CarthageKit/FrameworkExtensions.swift [hunk 1]
```diff
@@ -262,6 +262,38 @@ extension NSURL {
...
+
+	public func appendingPathComponent(pathComponent: String) -> NSURL {
+		#if swift(>=2.3)
+			return URLByAppendingPathComponent(pathComponent)!
+		#else
+			return URLByAppendingPathComponent(pathComponent)
+		#endif
+	}
+
+	public func appendingPathComponent(pathComponent: String, isDirectory: Bool) -> NSURL {
```
Comment (ikesyo) [inline Source/CarthageKit/FrameworkExtensions.swift]: These `appendingPath...` functions are used to work around optionality difference between Swift 2.2 and 2.3, and also match the Swift 3 APIs:

- https://github.com/apple/swift/blob/swift-3.0-PREVIEW-5/stdlib/public/SDK/Foundation/URL.swift#L784
- https://github.com/apple/swift/blob/swift-3.0-PREVIEW-5/stdlib/public/SDK/Foundation/URL.swift#L809
- https://github.com/apple/swift/blob/swift-3.0-PREVIEW-5/stdlib/public/SDK/Foundation/URL.swift#L855

**Comment by ikesyo** [comment]
Finally passed the test suites! It's ready for review. 🙏

File: Carthage.xcodeproj/xcshareddata/xcschemes/carthage.xcscheme [hunk 2]
```diff
@@ -77,6 +77,18 @@
...
+            key = "DEVELOPER_DIR"
+            value = "${DEVELOPER_DIR}"
+            isEnabled = "YES">
+         </EnvironmentVariable>
+         <EnvironmentVariable
+            key = "TOOLCHAINS"
+            value = "${TOOLCHAINS}"
+            isEnabled = "YES">
+         </EnvironmentVariable>
+      </EnvironmentVariables>
```
Comment (mdiep) [inline Carthage.xcodeproj/xcshareddata/xcschemes/carthage.xcscheme]: Should this be left in the schemes?

**Comment by mdiep** [comment]
Just one question. This looks great! Thanks for all your hard work on it!

File: Carthage.xcodeproj/xcshareddata/xcschemes/carthage.xcscheme (hunk 2 above)
Comment (ikesyo) [inline Carthage.xcodeproj/xcshareddata/xcschemes/carthage.xcscheme]: I added this for supporting the case that running tests through Xcode 8 but Xcode 7.3.1 is selected for `xcodebuild` (Command Line Tools) and the opposite. If this is added and enabled, the `xcodebuild` commands executed in Carthage tests would be the same version with the Xcode a user is launching.

Does that makes sense or should we remove it?

Comment (mdiep) [inline Carthage.xcodeproj/xcshareddata/xcschemes/carthage.xcscheme]: That makes sense. 👍🏻

**Comment by mdiep** [comment]
🎉

**Pull request merged**